Move "getting started" to entirely new document (#116)
cert-manager/istio-csr: docs/getting_started.md

# Getting Started With istio-csr

> This guide will run through installing and using istio-csr from scratch. We'll use [kind](https://kind.sigs.k8s.io/) to create a new cluster locally in Docker, but this guide should work on any cluster.

Reviewer: There are extra steps that must be done so that the cluster can be ready before insalling Istio. The Istio docs call these setup tasks "Plaform Setup" (add policies on openshift, add firewall rules and such). Should we mention these "Platform Setup" steps here?

```suggestion
This guide will run through installing and using istio-csr from scratch. We'll use [kind](https://kind.sigs.k8s.io/) to create a new cluster locally in Docker, but this guide should work on any cluster as long as the [Platform Setup](https://istio.io/latest/docs/setup/platform-setup/) has been performed. Note that if you are following the Platform Setup for OpenShift, do not run the `istioctl install` command listed in the Istio documentation; we will be performing later in this getting started guide.
```

Author: Good suggestion, thanks :+1:

```console
helm repo add jetstack https://charts.jetstack.io
helm repo update

# We set a few helm template values so we can point at our static root CA
helm install -n cert-manager cert-manager-istio-csr jetstack/cert-manager-istio-csr \
	--set "app.tls.rootCAFile=/var/run/secrets/istio-csr/ca.pem" \
	--set "volumeMounts[0].name=root-ca" \
	--set "volumeMounts[0].mountPath=/var/run/secrets/istio-csr" \
	--set "volumes[0].name=root-ca" \
	--set "volumes[0].secret.secretName=istio-root-ca"

# Check to see that the istio-csr pod is running and ready
kubectl get pods -n cert-manager
NAME                                       READY   STATUS    RESTARTS   AGE
cert-manager-aaaaaaaaaa-11111              1/1     Running   0          9m46s
cert-manager-cainjector-aaaaaaaaaa-22222   1/1     Running   0          9m46s
cert-manager-istio-csr-bbbbbbbbbb-00000    1/1     Running   0          63s
cert-manager-webhook-aaaaaaaaa-33333       1/1     Running   0          9m46s
```

## Installing Istio

If you're not running on kind, you may need to do some additional [setup tasks](https://istio.io/latest/docs/setup/platform-setup/) before installing Istio.

> This istio installation configuration is required for:
> 
> - The CA server to be disabled in istiod
> - Ensuring Istio workloads request certificates from istio-csr
> - Ensuring that the istiod certificates and keys are mounted from the Certificate created when installing istio-csr

Reviewer: I was a bit lost while reading here: what is the "installation configuration"?

```suggestion
We will be using the `istioctl` CLI to install Istio. In order for istio-csr to work, we will pass a custom IstioOperator manifest to `istioctl`. The custom manifest does the following:

- It disables the CA server in istiod,
- It makes sure that Istio workloads request certificates from istio-csr,
- It ensures that the istiod certificates and keys are mounted from the Certificate created when installing istio-csr.

Let us first download the custom IstioOperator manifest, and then use istioctl to install Istio:
```

Author: Much better wording yeah, thanks :+1:

```console
curl -sSL https://raw.githubusercontent.com/cert-manager/istio-csr/main/hack/istio-config-1.10.0.yaml > istio-install-config.yaml

# This takes a little time to complete
istioctl install -f istio-install-config.yaml

> ```

Reviewer: With OpenShift, the command varies slightly:

```sh
istioctl install --set profile=openshift -f <(curl -sSL https://raw.githubusercontent.com/SgtCoDFish/istio-csr/readmerewrite/hack/istio-config-1.10.0.yaml)
```

Could we somehow incorporate `--set profile=openshift` (for example, as a comment below the code block) for OpenShift users?

Author: done, thanks :+1:

kubectl get pods -n cert-manager
NAME                                       READY   STATUS    RESTARTS   AGE
cert-manager-aaaaaaaaaa-11111              1/1     Running   0          9m46s
cert-manager-cainjector-aaaaaaaaaa-22222   1/1     Running   0          9m46s
cert-manager-istio-csr-bbbbbbbbbb-00000    1/1     Running   0          63s
cert-manager-webhook-aaaaaaaaa-33333       1/1     Running   0          9m46s
```

## Installing Istio

If you're not running on kind, you may need to do some additional [setup tasks](https://istio.io/latest/docs/setup/platform-setup/) before installing Istio.

We use the `istioctl` CLI to install Istio, configured using a custom IstioOperator manifest.

The custom manifest does the following:

- Disables the CA server in istiod,
- Ensures that Istio workloads request certificates from istio-csr,
- Ensures that the istiod certificates and keys are mounted from the Certificate created when installing istio-csr.

First we download our demo manifest and then we apply it.

```console
curl -sSL https://raw.githubusercontent.com/cert-manager/istio-csr/main/hack/istio-config-1.10.0.yaml > istio-install-config.yaml

```

> You may wish to inspect and tweak `istio-isntall-config.yaml` if you know what you're doing,

Reviewer 2: ```suggestion
You may wish to inspect and tweak `istio-install-config.yaml` if you know what you're doing,
```
